### PR TITLE
test(core): add testing for staking

### PR DESCRIPTION
**Pull request by facundoCarreSimtlix**
add testing for staking using varifyTransaction and explatin transacction

STLX-10863

### CODE REVIEW

File: modules/core/test/v2/unit/coins/sol.ts [hunk 1]
```diff
@@ -165,6 +174,31 @@ describe('SOL:', function () {
       const validTransaction = await basecoin.verifyTransaction({ txParams, txPrebuild, memo, durableNonce });
       validTransaction.should.equal(true);
     });
+    it('should verify transactions with stakig ', async function () {
```
Comment (Krin2) [inline modules/core/test/v2/unit/coins/sol.ts:177]: staking

**Review by ipelaez-simtlix — CHANGES_REQUESTED** [review]
Please add a withdraw case since it has outputs also

File: modules/core/test/v2/unit/coins/sol.ts [hunk 2]
```diff
@@ -76,6 +77,13 @@ describe('SOL:', function () {
   };
   const errorMemo = 'different memo';
   const errorFeePayer = '5hr5fisPi6DXCuuRpm5XUbzpiEnmdyxXuBDTwzwZj5Pe';
+  const factory = accountLib.register('tsol', accountLib.Sol.TransactionBuilderFactory);
+  const wallet = new accountLib.Sol.KeyPair(resources.authAccount).getKeys();
+  const stakeAccount = new accountLib.Sol.KeyPair(resources.stakeAccount).getKeys();
+  const blockHash = resources.blockHashes.validBlockHashes[0];
```
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/coins/sol.ts:83]: magic number , replace as a variable.

File: modules/core/test/v2/unit/coins/sol.ts [hunk 3]
```diff
@@ -165,6 +174,55 @@ describe('SOL:', function () {
...
+      txParams.recipients = [
+        {
+          address: '7dRuGFbU2y2kijP6o1LYNzVyz4yf13MooqoionCzv5Za',
+          amount: amount,
+        },
+      ];
+      const validTransaction = await basecoin.verifyTransaction({ txParams, txPrebuild, memo });
+      validTransaction.should.equal(true);
+    });
+    it('should verify whitdraw staking transaction', async function () {
```
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/coins/sol.ts:202]: withdraw

File: modules/core/test/v2/unit/coins/sol.ts [hunk 4]
```diff
@@ -165,6 +174,55 @@ describe('SOL:', function () {
       const validTransaction = await basecoin.verifyTransaction({ txParams, txPrebuild, memo, durableNonce });
       validTransaction.should.equal(true);
     });
+    it('should verify activate staking transaction', async function () {
```
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/coins/sol.ts:177]: there is no test for deactivate builder. 
Also there are not failure cases. Please try to add failure cases.

File: modules/core/test/v2/unit/coins/sol.ts [hunk 5]
```diff
@@ -453,6 +511,58 @@ describe('SOL:', function () {
         memo: undefined,
       });
     });
+
+    it('should explain an unsigned transfer for staking', async function () {
```
Comment (rliberal-simtlix) [inline modules/core/test/v2/unit/coins/sol.ts:515]: also add explain unsigned withdraw and unsigned deactivate.

File: modules/core/test/v2/unit/coins/sol.ts (hunk 4 above)
Comment (facundoCarreSimtlix) [reply modules/core/test/v2/unit/coins/sol.ts:177]: This is covered from here "line 136", to here "line 171 "

**Review by rliberal-simtlix — APPROVED** [review]

**Review by ipelaez-simtlix — APPROVED** [review]

**Review by joey-bai — APPROVED** [review]

**Pull request merged**